src/udev/udev-builtin-input_id.c

```diff
@@ -193,7 +195,10 @@ static bool test_pointers(struct udev_device *dev,
         is_pointing_stick = test_bit(INPUT_PROP_POINTING_STICK, bitmask_props);
         stylus_or_pen = test_bit(BTN_STYLUS, bitmask_key) || test_bit(BTN_TOOL_PEN, bitmask_key);
         finger_but_no_pen = test_bit(BTN_TOOL_FINGER, bitmask_key) && !test_bit(BTN_TOOL_PEN, bitmask_key);
-        has_mouse_button = test_bit(BTN_LEFT, bitmask_key);
+
+        /* Set has_mouse_button if any button in [BTN_MOUSE, BTN_JOYSTICK) is found */
+        for (button = BTN_MOUSE; button < BTN_JOYSTICK && !(has_mouse_button = test_bit(button, bitmask_key)); button++);
```

Comment:
can you please follow coding style and use `;` on a separate line if you do for loops with no body?

Also, I really don't appreciate this overly condensed style I must say, this doesn't help readability. Instead of cramming the entire logic with variable assignment and everything into the second part of the for description, why not just place it in the body of the for loop and make it actually readable?

This is not a contest in writing the shortest most unreadable code, but quite the opposite: the focus is on readability.